skiko/src/awtMain/kotlin/org/jetbrains/skiko/SkiaLayer.awt.kt

```diff
@@ -133,6 +135,16 @@ actual open class SkiaLayer internal constructor(
         @Suppress("LeakingThis")
         add(backedLayer)
 
+        addAncestorListener(object : AncestorListener {
+            override fun ancestorAdded(event: AncestorEvent?) = Unit
+
+            override fun ancestorRemoved(event: AncestorEvent?) = Unit
+
+            override fun ancestorMoved(event: AncestorEvent?) {
+                revalidate()
```

Comment:
Why not `redrawer?.syncBounds()` here?

Reply:
`revalidate` sounds more expensive...

Reply:
I guess I'm just not sure I understand why revalidation is needed here... Only the (absolute) position of the layer changes, not the size, so layout should not change.

Maybe `SwingUtilities.invokeLater { redrawer?.syncBounds() }` ?

Reply:
We can't say that `invokeLater` is in sync with layout.

Actually I figured why there were resize glitches. `addAncestorListener` receives events asynchronously. I replaced it by `addHierarchyBoundsListener` which is called synchronously on the parent layout. We don't need `revalidate` then.

Also, I looked at the `revalidate` doc, and you are right, it is in't performant, as it actually revalidates the whole tree back to the root.

So, I removed it from the other place, I am not confident now to fix another very small issue (recalling `adjustSizeToContentScale` on content scale change).

Reply:
Sorry, wait a minute. I tested it with a wrong version. Glitches are back. Investigating

Reply:
I returned to `revalidate`.

I looked at the documentation:
https://docs.oracle.com/javase/7/docs/api/javax/swing/JComponent.html#revalidate()

- it notify the `validateRoot`, but it doesn't validate all children, just what was `invalidated`
- it seems that only `validate` is expensive, but it is deferred, and actually is already called on every parent size change or when a component is added/removed

I don't want to user `invokeLater` because:
- we need to handle a proper cancelation during `dispose`
- we need to handle double `invokeLater`
- we are not sure that it will be called the same time as the layout. I investigated resize glitches, and it appeared that we called `syncBounds` too often - we called it even if it haven't performed `doLayout` yet, and so `syncBounds` was based on incorrect size.